### PR TITLE
DOC-2098: Region awareness Doc

### PR DESCRIPTION
**Pull request by akiyax**
Video https://drive.google.com/file/d/1-YR_V3crvFM_nrE0hm5S5zj7PFSRie5b/view

### CODE REVIEW

File: modules/cluster-and-ha-management/pages/expand-a-cluster.adoc [hunk 1]
```diff
@@ -94,6 +94,32 @@ If you choose to supply a staging location, make sure that the TigerGraph Linux
...
++
+--
+[source,console]
+----
+$ gadmin cluster expand m3:192.168.1.3,m4:192.168.1.4 --region-aware=false
+----
+--
+====
+
+If you do not need to configure region aware, you do not need to specify `region-aware` option, and the option will be consistent with the current cluster.
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/expand-a-cluster.adoc:121]: - Let's give a warning that cross region will have extended latency for queries.
- Let's add a comparison statement about the CRR feature. The pros and cons, or goals, of each. 
- We should also mention zones together with regions. Specifically, we should replace all the region with region/zone. Zones are typically closer geographically and actually preferable to be used. See more discussion here: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html

File: modules/cluster-and-ha-management/pages/ha-cluster.adoc [hunk 2]
```diff
@@ -8,6 +8,11 @@
 
 TigerGraph HA service provides load balancing when all components are operational, as well as automatic fail over in the event of a service disruption.
 
+[TIP]
+====
+The high available cluster can tolerate the failure of **a single node**. To tolerate **multiple node** failures, please refer to xref:region-aware.adoc[Region Aware Cluster Configuration].
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/ha-cluster.adoc]: This is not true, after region aware feature, we can tolerate more than one node down even for non-region aware case (if HA is on). Pls verify and confirm. What is special in region-aware deployment is that it can tolerate a **whole region/zone** outage.

File: modules/cluster-and-ha-management/pages/ha-overview.adoc [hunk 3]
```diff
@@ -58,6 +60,10 @@ This issue is no longer present in versions 3.9.3 and 3.10.0.
 
 Here you will find detailed information about terminology, system requirements, and how to configure an HA cluster.
 
+== xref:region-aware.adoc[Region Aware Cluster Configuration].
+
+Learn how to configure a high available cluster that can tolerate multiple node failures.
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/ha-overview.adoc]: ditto, see above

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 4]
```diff
@@ -0,0 +1,198 @@
+= Region Awareness
+//:page-aliases: tigergraph-server:region-aware:region-aware.adoc, tigergraph-server:region-aware:index.adoc
+:description: Overview of region awareness for TigerGraph servers.
+
+Region awareness represents a more sophisticated High Availability (HA) configuration within TigerGraph servers. In contrast to xref:ha-overview.adoc[High Availability], region-aware clusters offer HA at the **region level** rather than the **individual node level**.
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/region-aware.adoc]: This is accurate but contradicts with the earlier statements (the ones I left comment on).

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+There are two ways to enable region aware cluster:
+
+* Configure HA settings xref:installation:bare-metal-install.adoc#_step_2_configure_installation_settings[during installation].
+
+* xref:expand-a-cluster.adoc[Expand] or xref:shrink-a-cluster.adoc[Shrink a Cluster] after installation.
+
+
+
+=== During installation
+Configuring a region-aware cluster is part of platform xref:installation:bare-metal-install.adoc[installation].
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/region-aware.adoc:115]: We need a discussion about the format of region info.

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 6]
```diff
@@ -0,0 +1,198 @@
...
+$ gadmin config apply -y
+----
+
+====
+
+If you just want to enable region-aware without any other change, use the command following:
+
+[source,console]
+----
+$ gadmin cluster expand --region-aware
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/region-aware.adoc:206]: Does this work?

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 7]
```diff
@@ -0,0 +1,198 @@
...
+$ gadmin cluster expand us-east-3:m3:192.168.0.3,us-east-4:m4:192.168.0.4 --ha 2 --region-aware
+----
+
+== Disable Region Aware
+
+Disable region-awareness via xref:expand-a-cluster.adoc[Cluster Expand]
+
+[source,console]
+----
+$ gadmin cluster expand --region-aware=false
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/region-aware.adoc:222]: Does this work? And, do we really the cluster expansion in this case? Or it is a no-op?

File: modules/cluster-and-ha-management/pages/remove-failed-node.adoc [hunk 8]
```diff
@@ -52,7 +57,7 @@ NOTE: If you experience multi-node failures without modifying the default config
 You can check service status by running xref:system-management:management-with-gadmin.adoc#_gadmin_status[`gadmin status -v`].
 If a service is down, and you cannot restart it, please https://tigergraph.zendesk.com/hc/en-us/[open a support ticket].
 * There is at least 1 complete instance of the data.
-In other words, at least one replica does have a failed node.
+In other words, at least one replica does not have a failed node.
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/remove-failed-node.adoc]: "one partition", actually it would more clear if we say we need at lease one live replica for each partition.

File: modules/cluster-and-ha-management/pages/ha-cluster.adoc (hunk 2 above)
Comment (akiyax) [reply modules/cluster-and-ha-management/pages/ha-cluster.adoc]: I mean this is not accurate. I want to mention is, HA cluster can tolerate **ANY** one node down.

File: modules/cluster-and-ha-management/pages/region-aware.adoc (hunk 6 above)
Comment (akiyax) [reply modules/cluster-and-ha-management/pages/region-aware.adoc:206]: Yes

File: modules/cluster-and-ha-management/pages/region-aware.adoc (hunk 7 above)
Comment (akiyax) [reply modules/cluster-and-ha-management/pages/region-aware.adoc:222]: Yes, we do expand process to achieve region-aware

File: modules/cluster-and-ha-management/pages/ha-cluster.adoc (hunk 2 above)
Comment (chengjie-qin) [reply modules/cluster-and-ha-management/pages/ha-cluster.adoc]: But we mentioned `To tolerate **multiple node** failures, please refer toxxx`, this indirectly implies the HA cluster cannot tolerate multiple node failures which is wrong.

File: modules/cluster-and-ha-management/pages/region-aware.adoc (hunk 7 above)
Comment (akiyax) [reply modules/cluster-and-ha-management/pages/region-aware.adoc:222]: It will recover config to non-region-aware setup. Cloud will use this operation since the price for (non-)region-aware is different.

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 9]
```diff
@@ -59,7 +59,25 @@ By enabling region awareness, the service deployments would be structured as fol
...
+
+.Differences between Region-Aware and CRR
+|===
+|             | **Region-Aware**     |   **CRR**
+|Cluster Setup| 1 read-write   cluster  | 1 read-write cluster, other read-only cluster
+|Minimum nodes|  3                   | 2
+|Fail over    | Auto recover         | Manual switch
+|===
+
+Overall, if you are looking for zero downtime region failure toleration, use region-awareness. If you are looking for data residency compliance or Blue/green deployment, use CRR.
```
Comment (chengjie-qin) [inline modules/cluster-and-ha-management/pages/region-aware.adoc]: Let's mention the region-aware deployment can have higher latencies here.

File: modules/cluster-and-ha-management/pages/expand-a-cluster.adoc [hunk 10]
```diff
@@ -4,7 +4,7 @@
 
 Expanding a cluster adds more nodes to the cluster.
 During the expansion, the cluster's data is redistributed across the expanded cluster.
-You can also change replication factor during an expansion.
+You can also change replication factor or enable  xref:region-aware.adoc[region-aware] during an expansion.
```
Comment (victorleeTG) [inline modules/cluster-and-ha-management/pages/expand-a-cluster.adoc]:
```suggestion
You can also change replication factor or enable  xref:region-aware.adoc[region-awareness] during an expansion.
```

File: modules/cluster-and-ha-management/pages/expand-a-cluster.adoc [hunk 11]
```diff
@@ -94,6 +94,32 @@ If you choose to supply a staging location, make sure that the TigerGraph Linux
 For example, assume you are expanding from a 6-node cluster with a replication factor of 2 and a partitioning factor of 3, to a 10-node cluster with a replication factor of 2 and a partitioning factor of 5, and the size of the data root folder on a node is 50GB.
 You would need more than `(1 + ceiling(3/5)) * 50) = 100 GB` of free space on the staging path.
 
+==== Config region aware
```
Comment (victorleeTG) [inline modules/cluster-and-ha-management/pages/expand-a-cluster.adoc]:
```suggestion
==== Configure region awareness
```

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 12]
```diff
@@ -0,0 +1,226 @@
+= Region Awareness
+//:page-aliases: tigergraph-server:region-aware:region-aware.adoc, tigergraph-server:region-aware:index.adoc
+:description: Overview of region awareness for TigerGraph servers.
+
+Region awareness represents a more sophisticated High Availability (HA) configuration within TigerGraph servers. In contrast to xref:ha-overview.adoc[High Availability], region-aware clusters offer HA at the **region/zone level** rather than the **individual node level**.
```
Comment (victorleeTG) [inline modules/cluster-and-ha-management/pages/region-aware.adoc]:
```suggestion
Region awareness represents a more sophisticated High Availability (HA) configuration within TigerGraph servers. In contrast to standard xref:ha-overview.adoc[High Availability] clusters, region-aware clusters offer HA at the **region/zone level** rather than the **individual node level**.
```

File: modules/cluster-and-ha-management/pages/region-aware.adoc [hunk 13]
```diff
@@ -0,0 +1,226 @@
...
+
+Enabling region awareness involves deploying TigerGraph services based on distinct regions/zones. For instance, consider a 10-node cluster with 2 replica spanning 3 regions/zones:
```
Comment (victorleeTG) [inline modules/cluster-and-ha-management/pages/region-aware.adoc]:
```suggestion
Enabling region awareness involves deploying TigerGraph services based on distinct regions/zones. For instance, consider a 10-node cluster with 2 replicas spanning 3 regions/zones:
```

**Review by victorleeTG — COMMENTED** [review]
A lot of improvements for grammar and clarity of wording need to be made. But we will revise these after release.
